src/API.jl

```diff
@@ -959,9 +962,81 @@ function precompile(ctx::Context; internal_call::Bool=false)
             !internal_call && @warn "Circular dependency detected. Precompilation skipped for $pkg"
         end
     end
-    
+
+    pkg_queue = Base.PkgId[]
+    failed_deps = Base.PkgId[]
+    skipped_deps = Base.PkgId[]
+
     print_lock = stdout isa Base.LibuvStream ? stdout.lock : ReentrantLock()
-    errored = false
+    first_started = Base.Event()
+    finished = false
+    should_exit = false
+    
+    t_print = @async begin
+        wait(first_started)
+        isempty(pkg_queue) && return
+        lock(print_lock) do 
+            printpkgstyle(ctx, :Precompiling, "project...")
+        end
+        t = Timer(0; interval=1/10)
+        anim_chars = ["◐","◓","◑","◒"]
+        i = 1
+        last_length = 0
+        while !should_exit
+            lock(print_lock) do
+                term_size = Base.displaysize(stdout)::Tuple{Int,Int}
+                num_deps_show = term_size[1] - 2
+                pkg_queue_show = if !finished && length(pkg_queue) > num_deps_show 
+                    last(pkg_queue, num_deps_show)
+                else
+                    pkg_queue
+                end
+                str = ""
+                if i > 1
+                    str *= "\e[$(last_length)A\e[1G\e[0J"
+                end
+                for dep in pkg_queue_show
+                    finished && was_recompiled[dep] && continue
+                    name = dep in direct_deps ? "  $(dep.name)" : "  \e[38;5;244m$(dep.name)\e[0m"
+                    if dep in failed_deps
+                        str *= string(name, " \e[38;5;160m✗\e[0m\n")
```

Comment:
Could we use one of the standard 16 colors here? Could perhaps use `Base.error_color` for the red?